fuse/src/main/java/tachyon/fuse/TachyonFuse.java

```diff
@@ -0,0 +1,152 @@
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.client.ClientContext;
+import tachyon.conf.TachyonConf;
+
+/**
+ * Main entry point to Tachyon-FUSE
+ */
+public final class TachyonFuse {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static TachyonConf sTachyonConf;
+
+  public static void main(String[] args) {
+    sTachyonConf = ClientContext.getConf();
+    final TachyonFuseOptions opts = parseOptions(args);
+    if (opts == null) {
+      System.exit(1);
+    }
+
+    final TachyonFuseFs fs = new TachyonFuseFs(sTachyonConf, opts);
+    final List<String> fuseOpts = opts.getFuseOpts();
+    fuseOpts.add("-odirect_io");
+
+    try {
+      fs.mount(Paths.get(opts.getMountPoint()), true, opts.isDebug(),
+          fuseOpts.toArray(new String[0]));
+    } finally {
+      fs.umount();
+    }
+  }
+
+  /**
+   * Parses CLI options
+   * @param args CLI args
+   * @return Tachyon-FUSE configuration options
+   */
+  private static TachyonFuseOptions parseOptions(String[] args) {
+    final Options opts = new Options();
+    final Option mntPoint = Option.builder("m")
+        .hasArg()
+        .required(false)
+        .longOpt("mount-point")
+        .desc("Path where tachyon-fuse should be mounted.")
+        .build();
+
+    final Option tachyonRoot = Option.builder("r")
+        .hasArg()
+        .required(false)
+        .longOpt("tachyon-root")
+        .desc("Path within tachyon that will be the root of the mount (e.g., /users/foo)")
+        .build();
+
+    final Option help = Option.builder("h")
+        .required(false)
+        .desc("Print this help")
+        .build();
+
+    final Option fuseOption = Option.builder("o")
+        .valueSeparator(',')
+        .required(false)
+        .hasArgs()
+        .desc("FUSE mount options")
+        .build();
+
+    opts.addOption(mntPoint);
+    opts.addOption(tachyonRoot);
+    opts.addOption(help);
+    opts.addOption(fuseOption);
+
+    final CommandLineParser parser = new DefaultParser();
+    try {
+      CommandLine cli = parser.parse(opts, args);
+
+      if (cli.hasOption("h")) {
+        final HelpFormatter fmt = new HelpFormatter();
+        fmt.printHelp(TachyonFuse.class.getName(), opts);
+        return null;
+      }
+
+      String mntPointValue = cli.getOptionValue("m");
+      String tachyonRootValue = cli.getOptionValue("r");
+
+      List<String> fuseOpts = Lists.newArrayList();
+      boolean noUserMaxWrite = true;
+      if (cli.hasOption("o")) {
+        String[] fopts = cli.getOptionValues("o");
+        // keep the -o
+        for (int i = 0; i < fopts.length; i++) {
```

Comment:
can we simplify the for-loop as `for (fopt : fopts)`, since `i` is not really used rather than having `fopts[i]`?